Source/KSCrash/Recording/Tools/KSStackCursor_MachineContext.c

```diff
@@ -109,16 +109,23 @@ static bool advanceCursor(KSStackCursor *cursor)
         return false;
     }
     
-    if(context->instructionAddress == 0)
+    if(context->instructionAddress == 0 && cursor->state.currentDepth == 0)
     {
         context->instructionAddress = kscpu_instructionAddress(context->machineContext);
-        if(context->instructionAddress == 0)
-        {
-            return false;
-        }
         nextAddress = context->instructionAddress;
         goto successfulExit;
     }
+    
+    if(context->linkRegister == 0 && !context->isPastFramePointer)
+    {
+        // Link register, if available, is the second address in the trace.
+        context->linkRegister = kscpu_linkRegister(context->machineContext);
+        if(context->linkRegister != 0)
```

Comment:
What about this case?
> On an ARM Cortex M series device, the link register (LR or R14) is a core register that stores the return address, such as when making a function call.  In the case of an exception, the return address is pushed onto the stack by hardware and the LR is set to EXC_RETURN (0xFFFFFFF1, 0xFFFFFFF9, or 0xFFFFFFFD).
https://www.silabs.com/community/mcu/32-bit/knowledge-base.entry.html/2017/03/22/how_to_read_the_link-Z7Bq

Reply:
Yeah in that case I guess the second frame will be non-sense.. Any iOS device has this architecture(ARM Cortex M)? Cause I have never seen a crash report like that

Reply:
Well, I guess there are no such Apple devices, but as far as I get KSCrash is designed to work not only on these devices

Reply:
Realistically, 32-bit is going the way of the dodo, so I'm not too concerned with backwards compatibility anymore.